src/NHSD.BuyingCatalogue.Ordering.Services/SectionStatusService.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using NHSD.BuyingCatalogue.Ordering.Contracts;
+using NHSD.BuyingCatalogue.Ordering.Domain;
+using NHSD.BuyingCatalogue.Ordering.Persistence.Data;
+
+namespace NHSD.BuyingCatalogue.Ordering.Services
+{
+    public sealed class SectionStatusService : ISectionStatusService
+    {
+        private readonly ApplicationDbContext context;
+
+        public SectionStatusService(ApplicationDbContext context)
+        {
+            this.context = context ?? throw new ArgumentNullException(nameof(context));
+        }
+
+        public async Task<Order> GetOrder(CallOffId callOffId)
+        {
+            return await context.Order
+                .Where(o => o.Id == callOffId.Id)
+                .Include(o => o.Progress)
+                .SingleOrDefaultAsync();
+        }
+
+        public async Task SetSectionStatus(Order order, string sectionId)
+        {
+            if (order is null)
+                throw new ArgumentNullException(nameof(order));
+
+            if (sectionId is null)
+                throw new ArgumentNullException(nameof(sectionId));
+
+            var progressUpdate = sectionId switch
+            {
+                "catalogue-solutions" => order.Progress.CatalogueSolutionsViewed = true,
+                "additional-services" => order.Progress.AdditionalServicesViewed = true,
+                "associated-services" => order.Progress.AssociatedServicesViewed = true,
+                _ => throw new InvalidOperationException(),
+            };
```

Comment:
Although the `switch` expression syntax is nicer, in this case I think we should stick with a `switch` statement as we're not actually returning a value or using the `progressUpdate` variable. We could use a discard but I think it would be clearer just to use the `switch` statement.

Reply:
Used Switch statement.